sympy/functions/special/delta_functions.py

```diff
@@ -99,16 +101,21 @@ def simplify(self, x):
         """
         from sympy.polys.polyroots import roots
 
-        if not self.args[0].has(x) or (len(self.args) > 1 and self.args[1] != 0 ):
+        if 'wrt' not in hints:
+          raise TypeError("The 'wrt' keyword is required for the DiracDelta hint.")
```

Comment:
The case of "no free symbol" in an expression would be an  issue with that expression object only. Is it so?
And I feel that we have to add an if statement in order to check that whether an expression is passed as an argument through the Diracdelta object.